File: library/src/test/java/com/stealthcopter/networktools/PingTest.java
package com.stealthcopter.networktools;

import org.junit.Test;

/**
 * Created by matthew on 03/11/17.
 */

public class PingTest {

    @Test(expected = IllegalArgumentException.class)
    public void testIllegalArgumentThrownOnInvalidDelay() throws Exception {
        Ping.onAddress("127.0.0.1").setDelayMillis(-1);
    }

    @Test(expected = IllegalArgumentException.class)
    public void testIllegalArgumentThrownOnInvalidTimeout() throws Exception {
        Ping.onAddress("127.0.0.1").setTimeOutMillis(-1);
    }

    @Test(expected = IllegalArgumentException.class)
    public void testIllegalArgumentThrownOnInvalidTimes() throws Exception {
        Ping.onAddress("127.0.0.1").setTimes(-1);
    }

}

File: library/src/test/java/com/stealthcopter/networktools/PingTest.kt
package com.stealthcopter.networktools

import org.junit.Test

/**
 * Created by matthew on 03/11/17.
 */
class PingTest {
    @Test(expected = IllegalArgumentException::class)
    @Throws(Exception::class)
    fun testIllegalArgumentThrownOnInvalidDelay() {
        Ping.onAddress("127.0.0.1").setDelayMillis(-1)
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(Exception::class)
    fun testIllegalArgumentThrownOnInvalidTimeout() {
        Ping.onAddress("127.0.0.1").setTimeOutMillis(-1)
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(Exception::class)
    fun testIllegalArgumentThrownOnInvalidTimes() {
        Ping.onAddress("127.0.0.1").setTimes(-1)
    }
}